features/step_definitions/secrets.js

```diff
@@ -2,56 +2,70 @@
 
 const Assert = require('chai').assert;
 const request = require('../support/request');
+const { defineSupportCode } = require('cucumber');
+
 const TIMEOUT = 240 * 1000;
 
-module.exports = function server() {
-    this.Before({
-        tags: ['@secrets']
-    }, () => {
+defineSupportCode(({ Before, Given, When, Then, After }) => {
+    Before({
+        tags: '@secrets'
+    }, function hook() {
         this.repoOrg = this.testOrg;
         this.repoName = 'functional-secrets';
         this.pipelineId = null;
         this.secretId = null;
     });
 
-    this.Given(/^an existing repository for secret with these users and permissions:$/,
-        { timeout: TIMEOUT }, table =>
-        this.getJwt(this.accessKey)
-        .then((response) => {
-            this.jwt = response.body.token;
-
-            return request({
-                uri: `${this.instance}/${this.namespace}/pipelines`,
-                method: 'POST',
-                auth: {
-                    bearer: this.jwt
-                },
-                body: {
-                    checkoutUrl: `git@github.com:${this.repoOrg}/${this.repoName}.git#master`
-                },
+    Given(/^an existing repository for secret with these users and permissions:$/,
+        { timeout: TIMEOUT }, function step(table) {
+            return this.getJwt(this.accessKey)
+            .then((response) => {
+                this.jwt = response.body.token;
+
+                return request({
+                    uri: `${this.instance}/${this.namespace}/pipelines`,
+                    method: 'POST',
+                    auth: {
+                        bearer: this.jwt
+                    },
+                    body: {
+                        checkoutUrl: `git@github.com:${this.repoOrg}/${this.repoName}.git#master`
+                    },
+                    json: true
```

Comment:
This stuff got moved here because [line 159](https://github.com/screwdriver-cd/screwdriver/pull/617/files#diff-2b66eab6945aba4b87fba63045f30482R159) uses the `secondJobId` that we find with this `GET` request.

It didn't need to be here in the past because it got set in another step and stuff didn't get cleaned up, but now cucumber rebuilds the world (`this`) object on each scenario, so `this.secondJobId` was undefined in that step.